src/Adapter/Module/Tab/ModuleTabManagementSubscriber.php

```diff
@@ -76,4 +78,20 @@ public function onModuleUninstall(ModuleManagementEvent $event)
     {
         $this->moduleTabUnregister->unregisterTabs($event->getModule());
     }
+
+    /**
+     * @param ModuleManagementEvent $event
+     */
+    public function onModuleEnable(ModuleManagementEvent $event)
+    {
+        $this->moduleTabRegister->registerTabs($event->getModule());
+    }
+
+    /**
+     * @param ModuleManagementEvent $event
+     */
+    public function onModuleDisable(ModuleManagementEvent $event)
+    {
+        $this->moduleTabUnregister->unregisterTabs($event->getModule());
```

Comment:
i saw some case of modules who does not use new admin tab registration , as described on https://devdocs.prestashop.com/1.7/modules/concepts/controllers/admin-controllers/tabs/ 

but install tabs on install() function with 
`$tab = new Tab();
....
$tab->add();
`

if you push this code on prod, maybe module devs may be aware that their Bo tabs may disappear and not reappear after a module desactivation/reactivation.

Reply:
sorry, i have not tested this second comment , but, i think that unregisterTabs then registerTabs , reset permissions given to controllers, so it may be a strange behavior for merchants having to reconfigure permissions after just desactivation/reactivation.

Reply:
well actually when you create the tabs manually with `$tab = new Tab(); .... $tab->add();` there are chances you don't use the `getTabs()` method which is used by `registerTabs/unregisterTabs` so it won't be removed
but what `registerTabs` is look into the module and search for Controllers so that it adds automatically invisible tabs and their permissions so that the controller are accessible

besides I don't think it matters if the permissions on a disabled module are removed, as long as they come back when you enable it later, and I added these event listener because without them the tabs of disabled module are still present even though they are not accessible

Reply:
mmm,
i may do more test to find what is behind unregisterTabs ,
calling `tabRepository->findByModule($module->get('name'));`

Reply:
you're right! since it fetches all the tab linked to the module it will delete them all
I'm gonna change this so that the are hidden rather than deleted, but i'm afraid it will show other tabs supposed to stay invisible

Reply:
I think I can't make this modif for the 1.7.5, to do this properly I would need the add a `disabled` or `visible` field on the Tab table
But it is beyond the scope of the bug fixes for 1.7.5
This should be managed in 1.7.6, I'm gonna create an issue for this and remove this modif from this PR

Reply:
Here is the issue for the future feature https://github.com/PrestaShop/PrestaShop/issues/11578

Reply:
alright I removed the listeners, I think the PR is ready to go